data_steward/utils/bq.py

```diff
@@ -1,30 +1,134 @@
+def get_table_schema(table_name):
+    """
+    A helper function to create big query SchemaFields for dictionary definitions.
+
+    Given the table name, reads the schema from the schema definition file
+    and returns a list of SchemaField objects that can be used for table
+    creation.
+
+    :param table_name:  the table name to get BigQuery SchemaField information
+        for.
+    :returns:  a list of SchemaField objects representing the table's schema.
+    """
+    fields = fields_for(table_name)
+
+    schema = []
+    for column in fields:
+        name = column.get('name')
+        field_type = column.get('type')
+        mode = column.get('mode')
+        description = column.get('description')
+        column_def = bigquery.SchemaField(name, field_type, mode, description)
+
+        schema.append(column_def)
+
+    return schema
+
+
+def create_tables(project_id=None, fq_table_names=None, exists_ok=False):
+    """
+    Create an empty table(s) in a project.
+
+    Relies on a list of fully qualified table names.  This is a list of
+    strings formatted as 'project-id.dataset-id.table-name`.  This will
+    allow the table to be created using the schema defined in a definition
+    file without requiring the user to read the file or submit the filepath.
+
+    :param project_id:  The project that will contain the created table.
+    :param fq_table_names: A list of fully qualified table names.
+    :param exists_ok: A flag to throw an error if the table already exists.
+        Defaults to raising an error if the table already exists.
+
+    :raises RuntimeError: a runtime error if table creation fails for any
+        table in the list.
+
+    :return: A list of created table objects.
+    """
+
+    if not project_id or not isinstance(project_id, str):
+        raise RuntimeError("Must specify the project to create the tables in")
+
+    if not fq_table_names:
+        raise RuntimeError(
+            "Must specify a list of fully qualified table names to create")
+
+    if not isinstance(fq_table_names, list):
+        if not isinstance(fq_table_names, str):
+            raise RuntimeError("fq_table_names expects a list of strings")
+
+        fq_table_names = list(fq_table_names)
+
+    client = get_client(project_id=project_id)
+
+    successes = []
+    failures = []
+    for table_name in fq_table_names:
+        schema = get_table_schema(table_name.split('.')[2])
+
+        try:
+            table = bigquery.Table(table_name, schema=schema)
+            table = client.create_table(table, exists_ok)
+        except (GoogleAPIError, OSError, AttributeError, TypeError, ValueError):
+            LOGGER.exception('Unable to create table %s', table_name)
+            failures.append(table_name)
+        else:
+            successes.append(table)
+
+    if failures:
+        raise RuntimeError(f"Unable to create tables: {failures}")
+
+    return successes
+
+
+def query(query_str, project_id=None, use_cache=False):
```

Comment:
we can certainly improve by storing the client and supplying it if it already exists so we're not creating multiple clients.  the big thing i want to address is allowing people to fall back on relying on the environment variable all the time.  i don't like it.  it's too easy to mess up and not reset your environment variables before running something.  i think it's an accident waiting to happen.  by requiring them to get a client that expects a project_id to be defined, i hope it will enforce more thoughtful development practices.  suggestions?

Reply:
I just wanted to add a doc string to this and have it return a result, not a dataframe (which we have discussed in an eng sync).

Reply:
I can definitely employ that access pattern here.  Not an issue.